### PR TITLE
Add a Context Trait and update TransactionContext to be ZmqTransactionContext

### PR DESCRIPTION
**Pull request by agunde406**
Also updates the methods to use references as well as fix the way get_state is handled for multiple addresses.

### CODE REVIEW

File: src/processor/handler.rs [hunk 1]
```diff
@@ -181,12 +181,12 @@ pub trait Context {
 }
 
 #[derive(Clone)]
-pub struct TransactionContext {
+pub struct ZmqContext {
```
Comment (peterschwarz) [inline src/processor/handler.rs]: I'm not sure if this should be public anymore, since SDK users should not be creating one of these directly

Comment (agunde406) [reply src/processor/handler.rs]: Done

File: src/processor/handler.rs [hunk 2]
```diff
@@ -163,253 +151,32 @@ impl From<ReceiveError> for ContextError {
...
-                String::from("Status was not set for TpStateSetResponse"),
-            )),
-        }
-    }
+pub trait TransactionContext {
+    #[deprecated(
+        since = "0.3.0",
+        note = "please use `get_state_entry`  or  `get_state_entries` instead"
+    )]
+    fn get_state(&self, addresses: &[String]) -> Result<Vec<(String, Vec<u8>)>, ContextError>;
```
Comment (peterschwarz) [inline src/processor/handler.rs]: You can give this a default implementation

File: src/processor/handler.rs [hunk 3]
```diff
@@ -163,253 +151,32 @@ impl From<ReceiveError> for ContextError {
...
-    /// * `addresses` - the addresses to fetch
-    #[allow(clippy::needless_pass_by_value)]
-    pub fn delete_state(
-        &mut self,
-        addresses: Vec<String>,
-    ) -> Result<Option<Vec<String>>, ContextError> {
-        let mut request = TpStateDeleteRequest::new();
-        request.set_context_id(self.context_id.clone());
-        request.set_addresses(RepeatedField::from_vec(addresses.clone()));
+    fn get_state_entry(&self, addresses: &str) -> Result<Option<Vec<u8>>, ContextError>;
```
Comment (peterschwarz) [inline src/processor/handler.rs]: These methods could keep the original doc comments, for now

File: src/processor/handler.rs [hunk 4]
```diff
@@ -163,253 +151,32 @@ impl From<ReceiveError> for ContextError {
...
-        request.set_context_id(self.context_id.clone());
-        request.set_data(Vec::from(data));
+    fn get_state_entries(
+        &self,
+        addresses: &[String],
+    ) -> Result<Vec<(String, Vec<u8>)>, ContextError>;
 
-        let serialized = request.write_to_bytes()?;
-        let x: &[u8] = &serialized;
+    fn set_state(&self, entries: HashMap<String, Vec<u8>>) -> Result<(), ContextError>;
```
Comment (vaporos) [inline src/processor/handler.rs]: set_state and delete_state should mirror what we are doing w/get_state_entry/entries. In addition to the naming, get_state_entries uses a Vec and not HashMap.

**Pull request merged**